From efe2ccf9f4fe84bdc7ff41ca5f3423e1e6d5639d Mon Sep 17 00:00:00 2001
From: Joseph Chen <chenjh@rock-chips.com>
Date: Tue, 5 Jun 2018 10:25:15 +0800
Subject: [PATCH] power: rk818-charger: otg switch to dc charger power when
 init

Without this, otg supply is always provided by PMIC and PMIC
can not enter charge mode, because charge or otg supply output
is exclusive, only one choice at one time.

Change-Id: I445c1ec6aac9520593bfb53aa7681352417f0543
Signed-off-by: Joseph Chen <chenjh@rock-chips.com>
---
 drivers/power/rk818_charger.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/power/rk818_charger.c b/drivers/power/rk818_charger.c
index b2a5033e80db..52d2d6ffe996 100644
--- a/drivers/power/rk818_charger.c
+++ b/drivers/power/rk818_charger.c
@@ -1394,6 +1394,12 @@ static void rk818_cg_init_charger_state(struct rk818_charger *cg)
 	rk818_cg_init_finish_sig(cg);
 	rk818_cg_set_chrg_param(cg, cg->dc_charger);
 	rk818_cg_set_chrg_param(cg, cg->usb_charger);
+
+	if (cg->otg_in && cg->dc_in && cg->pdata->power_dc2otg) {
+		CG_INFO("otg power from dc adapter\n");
+		rk818_cg_set_otg_power(cg, USB_OTG_POWER_OFF);
+	}
+
 	CG_INFO("ac=%d, usb=%d, dc=%d, otg=%d, 5v=%d\n",
 		cg->ac_in, cg->usb_in, cg->dc_in, cg->otg_in, cg->otg_pmic5v);
 }
-- 
2.35.3

